src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,778 @@
+﻿using System;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.System;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Automation.Peers;
+using TreeViewItemAutomationPeer = Microsoft.UI.Xaml.Automation.Peers.TreeViewItemAutomationPeer;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class TreeViewItem : ListViewItem
+	{
+		private const long c_dragOverInterval = 1000 * 10000;
+		private const string c_multiSelectCheckBoxName = "MultiSelectCheckBox";
+		private const string c_expandCollapseChevronName = "ExpandCollapseChevron";
+
+		private bool m_expansionCycled;
+		private CheckBox m_selectionBox;
+		private DispatcherTimer m_expandContentTimer;
+		private TreeView m_ancestorTreeView;
+		private UIElement m_expandCollapseChevron;
+
+		public TreeViewItem()
+		{
+			DefaultStyleKey = typeof(TreeViewItem);
+			SetValue(TreeViewItemTemplateSettingsProperty, new TreeViewItemTemplateSettings());
+		}
+
+		protected override void OnKeyDown(KeyRoutedEventArgs e)
+		{
+			var targetNode = TreeNode;
+			if (targetNode != null)
+			{
+				var treeView = AncestorTreeView;
+				var key = e.Key;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'e' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)